content/blog/writing-crossguard-policies/index.md

```diff
@@ -0,0 +1,167 @@
+    validate: (
+        resources: q.Resolved<TResource>[],
+        args: StackValidationArgs,
+        reportViolation: ReportViolation) => Promise<void> | void,
+): StackValidation {
+    return (args: StackValidationArgs, reportViolation: ReportViolation) => {
+        const isInstance = (<any>resourceClass).isInstance;
+        if (!isInstance || typeof isInstance !== "function") {
+            return;
+        }
+        const resources = args.resources
+            .filter(r => isInstance({ __pulumiType: r.type }) === true)
+            .map(r => ({ ...r.props, urn: r.urn } as q.Resolved<TResource>));
+        validate(resources, args, reportViolation);
+    };
+}
+```
+
+The above example checks whether logging is enabled on S3 buckets. If a bucket does not have a `loggings` property defined, it will be in violation of the policy unless the bucket is being used as a logging `targetBucket` of another bucket. The policy check is skipped if the bucket's `id` isn't available, meaning this policy will only run during updates when the resource `id`s are available.
+
+At the time of writing, there is no built-in helper for stack validation policies similar to `validateTypedResource` for resource validation policies, so we've included our own helper that can be used until a [built-in helper](https://github.com/pulumi/pulumi-policy/issues/158) is available.
+
+## Resource Validation vs. Stack Validation
+
+|                                | Resource Validation                   | Stack Validation                                     |
+|--------------------------------|---------------------------------------|------------------------------------------------------|
+| What does it check?            | Individual resources                  | All resources in the stack                           |
+| When is the check performed?   | Before resources are created/modified | After all stack resources have been created/modified |
+| What information is available? | Resource _input_ properties           | Resource _output_ properties                         |
+
+In summary:
+
+- Use resource validation policies when you want to check individual types of resources and block them from being created or modified, checking resource _input_ properties.
+
+- Use stack validation policies when you want to check the stack as a whole (all resources) after they've been created or modified, checking resource _output_ properties.
+
+## Trying It Out
+
+In this post, we've seen the different types of CrossGuard policies that can be written. To give it a try, [download and install Pulumi]({{< relref "/docs/get-started" >}}) and [get started with policy as code]({{< relref "/docs/get-started/crossguard" >}}). For examples of policies, check out [Pulumi CrossGuard policies for AWS (AWSGuard)](https://github.com/pulumi/pulumi-policy-aws) or our [policy examples](https://github.com/pulumi/examples/tree/master/policy-packs).
```

Comment:
<p>
And as always, if you'd like to share what policies you have come up with on your own you can discuss them on the [Pulumi Community Slack].